controllers/mutation_reconcile_looper.go

```diff
@@ -571,3 +487,47 @@ func getStructFieldValue(v *cue.Struct, field string) (string, error) {
 	}
 	return f.Value.String()
 }
+
+func (m *MutationReconcileLooper) getSource(ctx context.Context, ref v1alpha1.PatchStrategicMergeSourceRef) (sourcev1.Source, error) {
+	var sourceObj sourcev1.Source
+	key := types.NamespacedName{
+		Name:      ref.Name,
+		Namespace: ref.Namespace,
+	}
+	switch ref.Kind {
+	case sourcev1.GitRepositoryKind:
+		var repository sourcev1.GitRepository
+		err := m.Client.Get(ctx, key, &repository)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return sourceObj, err
+			}
+			return sourceObj, fmt.Errorf("unable to get source '%s': %w", key, err)
+		}
+		sourceObj = &repository
+	case sourcev1.BucketKind:
+		var bucket sourcev1.Bucket
+		err := m.Client.Get(ctx, key, &bucket)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return sourceObj, err
+			}
+			return sourceObj, fmt.Errorf("unable to get source '%s': %w", key, err)
+		}
+		sourceObj = &bucket
+	case sourcev1.OCIRepositoryKind:
+		var ocirepo sourcev1.OCIRepository
+		err := m.Client.Get(ctx, key, &ocirepo)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return sourceObj, err
+			}
+			return sourceObj, fmt.Errorf("unable to get source '%s': %w", key, err)
+		}
+		sourceObj = &ocirepo
+	default:
+		return sourceObj, fmt.Errorf("source `%s` kind '%s' not supported", ref.Name, ref.Kind)
+	}
+
+	return sourceObj, nil
+}
```

Comment:
```suggestion
func (m *MutationReconcileLooper) getSource(ctx context.Context, ref v1alpha1.PatchStrategicMergeSourceRef) (sourcev1.Source, error) {
	get := func(obj client.Object) error {
		key := types.NamespacedName{
			Name:      ref.Name,
			Namespace: ref.Namespace,
		}
		err := m.Client.Get(ctx, key, obj)
		if err != nil {
			return fmt.Errorf("unable to get source '%s': %w", key, err)
		}
		return nil
	}

	var obj client.Object
	switch ref.Kind {
	case sourcev1.GitRepositoryKind:
		obj = &sourcev1.GitRepository{}
	case sourcev1.BucketKind:
		obj = &sourcev1.Bucket{}
	case sourcev1.OCIRepositoryKind:
		obj = &sourcev1.OCIRepository{}
	default:
		return nil, fmt.Errorf("source `%s` kind '%s' not supported", ref.Name, ref.Kind)
	}

	if err := get(obj); err != nil {
		return nil, fmt.Errorf("unable to get source '%s': %w", ref, err)
	}

	return obj.(sourcev1.Source), nil
}
}
```

Reply:
What do you think? :)

Reply:
At this point you can even remove the `get` and just have the `m.Client.Get`. Pretty cool. :)

Reply:
Really nice 😄